File: Programmers/Level 1/비밀지도/Solution.py
def solution(n, arr1, arr2):
    answer = decode_map(to_binary(n, arr1), to_binary(n, arr2))
    return answer

def to_binary(n, arr):
    result = []
    for a in arr:
        b = bin(a)[2:]
        if len(b) < n:
            result.append("0"*(n-len(b)) + b)
        else:
            result.append(b)
    return result

def decode_map(arr1, arr2):
    map = []
    for i in range(len(arr1)):
        row = ""
        for j in range(len(arr1)):
            if arr1[i][j] == '1' or arr2[i][j] == '1':
                row += "#"
            else:
                row += " "
        map.append(row)
    return map

File: Programmers/Level 1/비밀지도/Solution.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<string> solution(int n, vector<int> arr1, vector<int> arr2) {
    vector<string> answer(n,"");
    
    for(int i=0;i<n;i++){
        for(int j=0;j<n;j++){
            int first = arr1[i]%2;
            int second = arr2[i]%2;
            
            if(first||second)
                answer[i]+='#';
            else
                answer[i]+=' ';
            
            arr1[i] /= 2;
            arr2[i] /= 2;
        }
        reverse(answer[i].begin(),answer[i].end());
    }
    
        
    return answer;
}